jenkins.sh

```diff
@@ -10,7 +10,41 @@ pip2.7 install sphinx
 make develop extras=[aws,mesos,azure,encryption,cwl]
 export LIBPROCESS_IP=127.0.0.1
 export PYTEST_ADDOPTS="--junitxml=test-report.xml"
-rm -rf /mnt/ephemeral/tmp
-mkdir /mnt/ephemeral/tmp && export TMPDIR=/mnt/ephemeral/tmp
-make $make_targets
-rm -rf /mnt/ephemeral/tmp
+export make_targets
+
+bash -ex <<-"END"
+    # If there is an output directory, delete it.  Furthermore, if the directory is a mountpoint
+    # then it must be on the loopback filesystem and needs to be unmounted.  Lastly, delete the
+    # image used for the loopback system.
+    cleanup() {
+        trap - EXIT
+        # Delete the temp dir
+        if [ -d /mnt/ephemeral/tmp ]
+        then
+            if mountpoint -q /mnt/ephemeral/tmp
+            then
+                # If there are processes still interacting with the device, kill them before
+                # unmounting
+                lsof +D /mnt/ephemeral/tmp -t | xargs -r -n1 kill
+                sudo umount /dev/loop0
+            fi
+        fi
+        rm -rf /mnt/ephemeral/tmp*
```

Comment:
Mention each file/directory explicitly. The wildcard could delete things that don't belong to you.